src/platform/Zephyr/DiagnosticDataProviderImpl.cpp

```diff
@@ -212,7 +212,30 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetNetworkInterfaces(NetworkInterface **
             ifp->hardwareAddress = ByteSpan(ifp->MacAddress, addressSize);
         }
 
-        head = ifp;
+        // Assuming IPv6-only support
+        Inet::InterfaceAddressIterator interfaceAddressIterator;
+        chip::Inet::IPAddress ipv6Addresses[kMaxIPv6AddrCount];
```

Comment:
You don't need an array here. A single object will be enough since you immediately copy its contents to another array.

Reply:
Right, fixed